tests/TestCase/ExceptionsTest.php

```diff
@@ -166,13 +166,13 @@ public function exceptionProvider()
             ['Cake\Database\Exception\MissingDriverException', 500],
             ['Cake\Database\Exception\MissingExtensionException', 500],
             ['Cake\Database\Exception\NestedTransactionRollbackException', 500],
-            ['Cake\Datasource\Exception\InvalidPrimaryKeyException', 404],
+            ['Cake\Datasource\Exception\InvalidPrimaryKeyException', 500],
             ['Cake\Datasource\Exception\MissingDatasourceConfigException', 500],
             ['Cake\Datasource\Exception\MissingDatasourceException', 500],
             ['Cake\Datasource\Exception\MissingModelException', 500],
             ['Cake\Datasource\Exception\PageOutOfBoundsException', 404],
-            ['Cake\Datasource\Exception\RecordNotFoundException', 404],
-            ['Cake\Mailer\Exception\MissingActionException', 404],
+            ['Cake\Datasource\Exception\RecordNotFoundException', 500],
```

Comment:
`RecordNotFoundException` behaviour is mentioned in [Getting a Single Entity by Primary Key](https://book.cakephp.org/4/en/orm/retrieving-data-and-resultsets.html#getting-a-single-entity-by-primary-key) docs and baked by [view.twig](https://github.com/cakephp/bake/blob/master/templates/bake/element/Controller/view.twig) and [edit.twig](https://github.com/cakephp/bake/blob/master/templates/bake/element/Controller/edit.twig), so response code should stay same and handled by `ExceptionRenderer` too.

Reply:
Jep, this is always a 404.

Reply:
I disagree, but I will implement the existing functionality.